ci/cloudbuild/builds/clang-tidy-full.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+#
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+source "$(dirname "$0")/../../lib/init.sh"
+source module ci/cloudbuild/builds/lib/cmake.sh
+source module ci/cloudbuild/builds/lib/features.sh
+source module ci/cloudbuild/builds/lib/integration.sh
+source module ci/lib/io.sh
+
+export CC=clang
+export CXX=clang++
+export CTCACHE_DIR=~/.cache/ctcache
+
+mapfile -t cmake_args < <(cmake::common_args)
+read -r ENABLED_FEATURES < <(features::list_full_cmake)
+readonly ENABLED_FEATURES
+
+# See https://github.com/matus-chochlik/ctcache for docs about the clang-tidy-cache
+# Note: we use C++14 for this build because we don't want tidy suggestions that
+# require a newer C++ standard.
+io::run cmake "${cmake_args[@]}" \
+  -DCMAKE_CXX_CLANG_TIDY=/usr/local/bin/clang-tidy-wrapper \
+  -DCMAKE_CXX_STANDARD=14 \
+  -DGOOGLE_CLOUD_CPP_ENABLE="${ENABLED_FEATURES}" \
+  -DGOOGLE_CLOUD_CPP_INTERNAL_DOCFX=ON
+io::run cmake --build cmake-out
+
+mapfile -t ctest_args < <(ctest::common_args)
+io::run env -C cmake-out ctest "${ctest_args[@]}" -LE integration-test
+
+integration::ctest_with_emulators "cmake-out"
+
+if [[ "${TRIGGER_TYPE}" != "manual" ]]; then
+  # This build should fail if any of the above work generated
+  # code differences (for example, by updating .bzl files).
+  io::log_h2 "Highlight generated code differences"
+  git diff --exit-code
+fi
```

Comment:
optional: I am not sure it is useful to run the integration tests here, or verify the generated code matches. We do these things in other builds.